plugin/pkg/shoot/validator/admission_test.go

```diff
@@ -1909,8 +1909,30 @@ var _ = Describe("validator", func() {
 					Entry("should allow when oidcConfig is valid", ptr.To("someClientID"), ptr.To("https://issuer.com"), BeNil()),
 					Entry("should forbid when oidcConfig clientID is nil", nil, ptr.To("https://issuer.com"), BeForbiddenError()),
 					Entry("should forbid when oidcConfig clientID is empty string", ptr.To(""), ptr.To("https://issuer.com"), BeForbiddenError()),
-					Entry("should forbid when oidcConfig IssuerURL is nil", ptr.To("someClientID"), nil, BeForbiddenError()),
-					Entry("should forbid when oidcConfig IssuerURL is empty string", ptr.To("someClientID"), ptr.To(""), BeForbiddenError()),
+					Entry("should forbid when oidcConfig issuerURL is nil", ptr.To("someClientID"), nil, BeForbiddenError()),
+					Entry("should forbid when oidcConfig issuerURL is empty string", ptr.To("someClientID"), ptr.To(""), BeForbiddenError()),
+				)
+
+				DescribeTable("do not validate oidc config when operation is not create", func(admissionOperation admission.Operation, operationOptions runtime.Object) {
+					oldShoot := shoot.DeepCopy()
+					shoot.Spec.Kubernetes.KubeAPIServer.OIDCConfig = &core.OIDCConfig{
+						ClientID:  ptr.To("someClientID"),
+						IssuerURL: ptr.To("https://issuer.com"),
+					}
```

Comment:
I rather meant to cover the case where old Shoot has `.spec.kubernetes.kubeAPIServer.oidcConfig={}` and the new Shoot has a other change like finalizer removal or annotation addition.
Because right now if we remove https://github.com/gardener/gardener/pull/10461/files#diff-279968515ef9a63760ad3936541fcf055a647877460573be64e3e1a87703c92bR726-R728 the test will still pass, right?

Wouldn't that make more sense?

Reply:
Resolved with https://github.com/gardener/gardener/pull/10461/commits/1ac74769f8570c3729f3dce56d3499146c537097.